src/kernel/pagecache.c

```diff
@@ -453,7 +478,7 @@ closure_function(5, 1, void, pagecache_write_sg_finish,
 
     pagecache_lock_node(pn);
     pagecache_page pp = page_lookup_nodelocked(pn, pi);
-    if (bound(complete)) {
+    if (!is_ok(s) || bound(complete)) {
```

Comment:
Really minor, but I would update the debug below to indicate whether the error was on write (complete true) or read. Yet I would still record a read error in pv->write_error, for it is an error for the cache write operation as a whole.